File: game-server/src/main/java/com/kingston/mmorpg/game/scene/director/BaseDungeonDirector.java
package com.kingston.mmorpg.game.scene.director;

import java.util.HashSet;
import java.util.Set;

/**
 * 副本默认场景管理器
 * 
 * @author kingston
 *
 */
public abstract class BaseDungeonDirector extends SceneDirector {

	protected Set<Long> histroyPlayerIds = new HashSet<>();

	protected Set<Long> alivePlayerIds = new HashSet<>();

	public abstract void init();

	public abstract void prepare();

	public abstract void start();

	public abstract void tryFinish();

	public void end() {
		settlement();
	}

	public abstract void settlement();

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/scene/director/BaseDungeonDirector.kt
package org.forfun.mmorpg.game.scene.director

/**
 * 副本默认场景管理器
 */
abstract class BaseDungeonDirector : SceneDirector() {

    protected val historyPlayerIds = HashSet<Long>()

    protected val alivePlayerIds = HashSet<Long>()

    abstract fun init()

    abstract fun prepare()

    abstract fun start()

    abstract fun tryFinish()

    fun end() {
        settlement()
    }

    abstract fun settlement()
}
